src/Symfony/Component/Messenger/Bridge/Kafka/Transport/KafkaReceiver.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Messenger\Bridge\Kafka\Transport;
+
+use Psr\Log\LoggerInterface;
+use RdKafka\KafkaConsumer;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Exception\TransportException;
+use Symfony\Component\Messenger\Transport\Receiver\ReceiverInterface;
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
+
+/**
+ * @author Konstantin Scheumann <konstantin@konstantin.codes>
+ */
+class KafkaReceiver implements ReceiverInterface
+{
+    /** @var LoggerInterface */
+    private $logger;
+
+    /** @var SerializerInterface */
+    private $serializer;
+
+    /** @var RdKafkaFactory */
+    private $rdKafkaFactory;
+
+    /** @var KafkaReceiverProperties */
+    private $properties;
+
+    /** @var KafkaConsumer */
+    private $consumer;
+
+    /** @var bool */
+    private $subscribed;
+
+    public function __construct(
+        LoggerInterface $logger,
+        SerializerInterface $serializer,
+        RdKafkaFactory $rdKafkaFactory,
+        KafkaReceiverProperties $properties
+    ) {
+        $this->logger = $logger;
+        $this->serializer = $serializer;
+        $this->rdKafkaFactory = $rdKafkaFactory;
+        $this->properties = $properties;
+
+        $this->subscribed = false;
+    }
+
+    public function get(): iterable
+    {
+        $message = $this->getSubscribedConsumer()->consume($this->properties->getReceiveTimeoutMs());
+
+        switch ($message->err) {
+            case \RD_KAFKA_RESP_ERR_NO_ERROR:
+                $this->logger->info(sprintf(
+                    'Kafka: Message %s %s %s received ',
+                    $message->topic_name,
+                    $message->partition,
+                    $message->offset
+                ));
+
+                $envelope = $this->serializer->decode([
+                    'body' => $message->payload,
+                    'headers' => $message->headers,
+                    'key' => $message->key,
+                    'partition' => $message->partition,
+                    'offset' => $message->offset,
+                    'timestamp' => $message->timestamp,
+                ]);
+
+                return [$envelope->with(new KafkaMessageStamp($message))];
+            case \RD_KAFKA_RESP_ERR__PARTITION_EOF:
+                $this->logger->info('Kafka: Partition EOF reached. Waiting for next message ...');
```

Comment:
@jderusse it will be sent once consumer reaches and of partition, which basically means that no further messages are waiting to be processed by this particular consumer in this particular partition of a topic.

https://docs.confluent.io/2.0.0/clients/librdkafka/rdkafka_8h.html#a03509bab51072c72a8dcf52337e6d5cbaffd61f12a5298f508039a9e2521752a5
RD_KAFKA_RESP_ERR__PARTITION_EOF | Reached the end of the topic+partition queue on the broker. Not really an error.
-- | --

I agree that this will flood the logs usually and should not be logged, or it could be moved to a `debug` level.